src/components/CodeSnippet/CodeSnippet.astro

```diff
@@ -318,3 +352,26 @@ function parseMarkingDefinition(serializedArr: string, parts: RegExp, parseError
 	<figcaption class="header">{titleHtml && <span class="title" set:html={titleHtml} />}</figcaption>
 	<Fragment set:html={codeSnippetHtml} />
 </figure>
+<script>
+	// Find all buttons with the \`copy-button\` class on page.
+	const buttons = document.querySelectorAll('button.copy-button') as NodeListOf<HTMLButtonElement>;
+	
+	// Handle clicks on each button.
+	buttons.forEach((button) => {
```

Comment:
The “copied” pop-up like in the implementation I linked to is important because it allows us to use an [ARIA Live Region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) to announce the successful action to assistive technologies like screen readers.

Here’s an example of how VoiceOver on macOS reads the interaction on the https://astro.build homepage (no sound but you can see what is read out in the box on the right):

https://user-images.githubusercontent.com/357379/227250731-ea585c0a-dccc-43f2-aa3b-dd65e6ec12b2.mp4

Reply:
I'll need a hand implementing this. 🙏 😅